### ISSUE TITLE
Cannot reproduce KNN performance for vanilla ViT-S training

### ISSUE THREAD

**Issue by wangh09** [facebookresearch/dino#280]
Hi! I followed the instructions in the readme.md to train a vanilla ViT-S model, and yield 68.4% KNN top1 acc which is 0.9% less than the claimed one. Is it because pytorch version matters or I should export the teacher backbone for evaluation? Thanks!

**Comment by wangh09** [comment]
ok loss doesn't match perfectly so I suppose this should due to version/GPU mismatch. I'm using V100*8, torch 1.13, cuda 11.7.99 btw
```
{"train_loss": 7.292225784392094, "train_lr": 4.99820136696111e-05, "train_wd": 0.04002958960483863, "epoch": 0}
{"train_loss": 9.178665082600954, "train_lr": 0.00014998601063191967, "train_wd": 0.04020716317255338, "epoch": 1}
{"train_loss": 7.321160263675961, "train_lr": 0.00024999000759422837, "train_wd": 0.0405621705635646, "epoch": 2}
{"train_loss": 5.968297207812897, "train_lr": 0.0003499940045565369, "train_wd": 0.04109426142843796, "epoch": 3}
{"train_loss": 5.297524073689009, "train_lr": 0.0004499980015188457, "train_wd": 0.04180291065772983, "epoch": 4}
{"train_loss": 4.919647876354906, "train_lr": 0.0005500019984811545, "train_wd": 0.04268741890020718, "epoch": 5}
{"train_loss": 4.628303665968535, "train_lr": 0.0006500059954434626, "train_wd": 0.043746913253022816, "epoch": 6}
{"train_loss": 4.389438543602717, "train_lr": 0.0007500099924057714, "train_wd": 0.04498034812316445, "epoch": 7}
{"train_loss": 4.199287116622848, "train_lr": 0.0008500139893680809, "train_wd": 0.046386506259330024, "epoch": 8}
{"train_loss": 4.043512085680481, "train_lr": 0.0009500179863303882, "train_wd": 0.047963999953206556, "epoch": 9}
{"train_loss": 3.897010138852419, "train_lr": 0.0009998986293786, "train_wd": 0.04971127240897121, "epoch": 10}
{"train_loss": 3.764632338385502, "train_lr": 0.0009992903106906335, "train_wd": 0.05162659927966087, "epoch": 11}
{"train_loss": 3.653709880477614, "train_lr": 0.0009980742928418459, "train_wd": 0.0537080903688947, "epoch": 12}
{"train_loss": 3.558310315787173, "train_lr": 0.0009962520573626758, "train_wd": 0.055953691496268755, "epoch": 13}
{"train_loss": 3.4752218733302693, "train_lr": 0.00099382582436636, "train_wd": 0.05836118652458501, "epoch": 14}
{"train_loss": 3.4072055228346354, "train_lr": 0.0009907985498440713, "train_wd": 0.060928199546911206, "epoch": 15}
{"train_loss": 3.3472413010091233, "train_lr": 0.0009871739220635018, "train_wd": 0.06365219723131352, "epoch": 16}
{"train_loss": 3.2930922461189716, "train_lr": 0.0009829563570752636, "train_wd": 0.06653049132094936, "epoch": 17}
{"train_loss": 3.245216342280332, "train_lr": 0.0009781509933326402, "train_wd": 0.06956024128705336, "epoch": 18}
{"train_loss": 3.2062256193632703, "train_lr": 0.0009727636854311343, "train_wd": 0.07273845713219447, "epoch": 19}
{"train_loss": 3.172901451671057, "train_lr": 0.0009668009969755688, "train_wd": 0.07606200234104478, "epoch": 20}
{"train_loss": 3.142900680871509, "train_lr": 0.0009602701925833546, "train_wd": 0.07952759697573956, "epoch": 21}
{"train_loss": 3.1184474723540143, "train_lr": 0.0009531792290336516, "train_wd": 0.08313182091278001, "epoch": 22}
{"train_loss": 3.094054794199556, "train_lr": 0.0009455367455732525, "train_wd": 0.08687111721828467, "epoch": 23}
{"train_loss": 3.0703399264388422, "train_lr": 0.0009373520533910319, "train_wd": 0.09074179565824961, "epoch": 24}
{"train_loss": 3.051779165804434, "train_lr": 0.0009286351242736563, "train_wd": 0.09474003634036816, "epoch": 25}
{"train_loss": 3.0355808223060947, "train_lr": 0.0009193965784565084, "train_wd": 0.09886189348379958, "epoch": 26}
{"train_loss": 3.0210863206621936, "train_lr": 0.0009096476716845882, "train_wd": 0.1031032993131822, "epoch": 27}
{"train_loss": 3.0088824124025595, "train_lr": 0.0008994002814990852, "train_wd": 0.10746006807303266, "epoch": 28}
{"train_loss": 3.0005448520374145, "train_lr": 0.0008886668927664516, "train_wd": 0.11192790015858395, "epoch": 29}
{"train_loss": 2.9869999627677277, "train_lr": 0.0008774605824675117, "train_wd": 0.11650238635897699, "epoch": 30}
{"train_loss": 2.9774344344671775, "train_lr": 0.0008657950037651597, "train_wd": 0.12117901220861477, "epoch": 31}
{"train_loss": 2.9678932948530816, "train_lr": 0.0008536843693701012, "train_wd": 0.12595316244240032, "epoch": 32}
{"train_loss": 2.9605710156482283, "train_lr": 0.0008411434342248507, "train_wd": 0.13082012555044326, "epoch": 33}
{"train_loss": 2.953843685327102, "train_lr": 0.0008281874775270916, "train_wd": 0.13577509842775098, "epoch": 34}
{"train_loss": 2.947069411178668, "train_lr": 0.0008148322841143536, "train_wd": 0.1408131911143144, "epoch": 35}
{"train_loss": 2.9387947707105693, "train_lr": 0.0008010941252325955, "train_wd": 0.14592943162090943, "epoch": 36}
{"train_loss": 2.931838548273968, "train_lr": 0.0007869897387122221, "train_wd": 0.15111877083584596, "epoch": 37}
{"train_loss": 2.926673547517386, "train_lr": 0.0007725363085755852, "train_wd": 0.15637608750783594, "epoch": 38}
{"train_loss": 2.9223563083760933, "train_lr": 0.000757751444100932, "train_wd": 0.16169619330004803, "epoch": 39}
{"train_loss": 2.9155170516811495, "train_lr": 0.0007426531583681869, "train_wd": 0.16707383791036914, "epoch": 40}
{"train_loss": 2.910682648086815, "train_lr": 0.0007272598463128065, "train_wd": 0.17250371425281952, "epoch": 41}
{"train_loss": 2.906146980327763, "train_lr": 0.0007115902623143696, "train_wd": 0.17798046369500614, "epoch": 42}
{"train_loss": 2.9015704989314175, "train_lr": 0.0006956634973472652, "train_wd": 0.18349868134644606, "epoch": 43}
{"train_loss": 2.897554629152532, "train_lr": 0.0006794989557212774, "train_wd": 0.18905292139253896, "epoch": 44}
{"train_loss": 2.891208848376259, "train_lr": 0.0006631163314404186, "train_wd": 0.19463770246893167, "epoch": 45}
{"train_loss": 2.887351329592492, "train_lr": 0.0006465355842088208, "train_wd": 0.20024751307096061, "epoch": 46}
{"train_loss": 2.8795733295565697, "train_lr": 0.000629776915112926, "train_wd": 0.20587681699284052, "epoch": 47}
{"train_loss": 2.873040037689735, "train_lr": 0.000612860742009547, "train_wd": 0.21152005879123734, "epoch": 48}
{"train_loss": 2.867980113704142, "train_lr": 0.0005958076746498755, "train_wd": 0.2171716692678112, "epoch": 49}
{"train_loss": 2.860328480839539, "train_lr": 0.0005786384895696651, "train_wd": 0.22282607096534862, "epoch": 50}
{"train_loss": 2.8528156104943543, "train_lr": 0.0005613741047762238, "train_wd": 0.2284776836720395, "epoch": 51}
{"train_loss": 2.844129244081503, "train_lr": 0.00054403555426305, "train_wd": 0.23412092992846917, "epoch": 52}
{"train_loss": 2.836523441185387, "train_lr": 0.0005266439623831464, "train_wd": 0.23975024053189964, "epoch": 53}
{"train_loss": 2.826469098671163, "train_lr": 0.0005092205181122464, "train_wd": 0.24536006003240118, "epoch": 54}
{"train_loss": 2.8138863444804763, "train_lr": 0.0004917864492333195, "train_wd": 0.25094485221540785, "epoch": 55}
{"train_loss": 2.804384565348629, "train_lr": 0.00047436299647378113, "train_wd": 0.2564991055652943, "epoch": 56}
{"train_loss": 2.793771407825293, "train_lr": 0.00045697138762694543, "train_wd": 0.262017338704579, "epoch": 57}
{"train_loss": 2.782490578927487, "train_lr": 0.00043963281168921483, "train_wd": 0.26749410580337685, "epoch": 58}
{"train_loss": 2.7704216660045797, "train_lr": 0.00042236839304457123, "train_wd": 0.27292400195378014, "epoch": 59}
{"train_loss": 2.7589406197567543, "train_lr": 0.00040519916572775066, "train_wd": 0.2783016685038454, "epoch": 60}
{"train_loss": 2.7448082663458315, "train_lr": 0.0003881460477975228, "train_wd": 0.2836217983459452, "epoch": 61}
{"train_loss": 2.731099476226323, "train_lr": 0.0003712298158512537, "train_wd": 0.28887914115423, "epoch": 62}
{"train_loss": 2.7164359804537654, "train_lr": 0.00035447107971182555, "train_wd": 0.29406850856607475, "epoch": 63}
{"train_loss": 2.70121210300141, "train_lr": 0.000337890257317738, "train_wd": 0.29918477930236326, "epoch": 64}
{"train_loss": 2.6858176910977285, "train_lr": 0.0003215075498469961, "train_wd": 0.30422290422157605, "epoch": 65}
{"train_loss": 2.6686244074055714, "train_lr": 0.0003053429171050933, "train_wd": 0.3091779113026757, "epoch": 66}
{"train_loss": 2.6505211185780073, "train_lr": 0.00028941605320705397, "train_wd": 0.3140449105518997, "epoch": 67}
{"train_loss": 2.6327532708204147, "train_lr": 0.0002737463625831922, "train_wd": 0.3188190988285892, "epoch": 68}
{"train_loss": 2.613737692727412, "train_lr": 0.00025835293633780243, "train_wd": 0.3234957645853076, "epoch": 69}
{"train_loss": 2.5945903835036486, "train_lr": 0.00024325452898958923, "train_wd": 0.3280702925175759, "epoch": 70}
{"train_loss": 2.573234539601109, "train_lr": 0.0002284695356221762, "train_wd": 0.33253816811861425, "epoch": 71}
{"train_loss": 2.5521876815018607, "train_lr": 0.0002140159694725254, "train_wd": 0.33689498213461816, "epoch": 72}
{"train_loss": 2.531404452620269, "train_lr": 0.0001999114399845916, "train_wd": 0.34113643491615664, "epoch": 73}
{"train_loss": 2.5090394447997606, "train_lr": 0.00018617313135491851, "train_wd": 0.3452583406614118, "epoch": 74}
{"train_loss": 2.4842466099752034, "train_lr": 0.00017281778159634602, "train_wd": 0.34925663154705805, "epoch": 75}
{"train_loss": 2.4618275195217247, "train_lr": 0.0001598616621453227, "train_wd": 0.3531273617427079, "epoch": 76}
{"train_loss": 2.4364056408667354, "train_lr": 0.00014732055803766083, "train_wd": 0.35686671130497855, "epoch": 77}
{"train_loss": 2.4109917147744664, "train_lr": 0.00013520974867690892, "train_wd": 0.36047098994731097, "epoch": 78}
{"train_loss": 2.3861272489900687, "train_lr": 0.00012354398921874443, "train_wd": 0.36393664068183257, "epoch": 79}
{"train_loss": 2.360169996811951, "train_lr": 0.00011233749259409383, "train_wd": 0.36726024332968277, "epoch": 80}
{"train_loss": 2.331089731463664, "train_lr": 0.00010160391219286703, "train_wd": 0.3704385178963119, "epoch": 81}
{"train_loss": 2.3052437658683003, "train_lr": 9.135632522940147e-05, "train_wd": 0.3734683278084384, "epoch": 82}
{"train_loss": 2.2763226497814144, "train_lr": 8.160721680989231e-05, "train_wd": 0.37634668300947466, "epoch": 83}
{"train_loss": 2.247880911834234, "train_lr": 7.236846472121322e-05, "train_wd": 0.3790707429103446, "epoch": 84}
{"train_loss": 2.218459681653195, "train_lr": 6.365132495966403e-05, "train_wd": 0.3816378191928037, "epoch": 85}
{"train_loss": 2.1912099093639497, "train_lr": 5.5466418017271704e-05, "train_wd": 0.38404537846249254, "epoch": 86}
{"train_loss": 2.161550723159199, "train_lr": 4.782371594235709e-05, "train_wd": 0.3862910447490771, "epoch": 87}
{"train_loss": 2.1331309113475823, "train_lr": 4.073253019013078e-05, "train_wd": 0.3883726018510539, "epoch": 88}
{"train_loss": 2.1055937971738126, "train_lr": 3.4201500278118214e-05, "train_wd": 0.39028799552287535, "epoch": 89}
{"train_loss": 2.0793594866633702, "train_lr": 2.8238583260238982e-05, "train_wd": 0.39203533550223313, "epoch": 90}
{"train_loss": 2.0546793335502285, "train_lr": 2.2851044032362814e-05, "train_wd": 0.39361289737553024, "epoch": 91}
{"train_loss": 2.030619491472376, "train_lr": 1.8045446481151825e-05, "train_wd": 0.39501912427965513, "epoch": 92}
{"train_loss": 2.009880992172243, "train_lr": 1.3827645486976646e-05, "train_wd": 0.39625262843843173, "epoch": 93}
{"train_loss": 1.9902538600668824, "train_lr": 1.0202779790646348e-05, "train_wd": 0.3973121925321758, "epoch": 94}
{"train_loss": 1.9724975991949474, "train_lr": 7.175265732643999e-06, "train_wd": 0.39819677089905287, "epoch": 95}
{"train_loss": 1.9589693277990885, "train_lr": 4.7487918724965946e-06, "train_wd": 0.39890549056700864, "epoch": 96}
{"train_loss": 1.9499998331058035, "train_lr": 2.9263144948330045e-06, "train_wd": 0.3994376521152962, "epoch": 97}
{"train_loss": 1.9430465311002578, "train_lr": 1.710054007606848e-06, "train_wd": 0.3997927303647176, "epoch": 98}
{"train_loss": 1.938800999598442, "train_lr": 1.1014922368712136e-06, "train_wd": 0.3999703748959069, "epoch": 99}
```

**Comment by wangh09** [comment]
I re-run vanilla training using PyTorch 1.7 (68.53%), PyTorch 1.13 (68.43%) PyTorch 2.3 (68.40%) without changing a line of code. Although the results show some correlation with PyTorch&CUDA versions but the difference should less than 0.2%. Is there any chance that there's a typo in the doc or the code for the 69.3% training is different with current main branch? Thanks!
